assert.sh

```diff
@@ -103,7 +103,7 @@ assert() {
     (( tests_ran++ )) || :
     [[ -z "$DISCOVERONLY" ]] || return
     expected=$(echo -ne "${2:-}")
-    result="$(eval 2>/dev/null $1 <<< ${3:-})" || true
+    result="$(eval 2>/dev/null "$1" <<< ${3:-})" || true
```

Comment:
Wrapping the command ($1) in quotes for both assert and assert_raises enables support for multiline input. See new tests added to test.sh to prove this.